WebdevPeriod3/wwwroot/js/subsystemlist.js

```diff
@@ -0,0 +1,115 @@
+﻿'use strict';
+
+function getButtonRow(buttonEl) { 
+  buttonEl.closest('input'); 
+}
+
+function debounce(callback, wait, immediate = false) {
+  let timeout = null 
+  
+  return function() {
+    const callNow = immediate && !timeout
+    const next = () => callback.apply(this, arguments)
+    
+    clearTimeout(timeout)
+    timeout = setTimeout(next, wait)
+
+    if (callNow) {
+      next()
+    }
+  }
+}
+
+let ctrl = null;
+
+async function search(term) {
+  if (ctrl) ctrl.abort();
+  ctrl = new AbortController();
+
+  const products = await fetch(`../products?searchterm=${term}`, {
+    signal: ctrl.signal
+  })
+    .then((res) => res.json())
+    .catch((ex) => (ex.name === "AbortError" ? null : Promise.reject(ex)))
+    .finally(() => {
+      ctrl = null;
+    });
+
+    console.log(products);
+
+  return products?.filter((product) =>
+    product.name.toLowerCase()
+  );
+}
+
+document.addEventListener("DOMContentLoaded", () => {
+
+  let terms;
+
+    document.getElementById("search-input").addEventListener("input", debounce(async (e) => {
+    if (e.target.value)
+		  terms = await search(e.target.value);
+		if (terms) {
+			document.getElementById("search-output").innerHTML = terms.slice(0,5).filter((term) => term.name != e.target.value).map((term) => `<li>${term.name}</li>`).join("");
+		}
+	}, 1000));
+
+  [].forEach.call(document.getElementById("search-output").children, (element) => { 
+    element.addEventListener('click', () => {
+      document.getElementById("search-input").value = this.value;
+    });
+  });
+
+  document.getElementById("search-output").addEventListener("click", (e) => {
+    document.getElementById("search-input").value = e.target.innerHTML;
+  })
+
+  document.getElementById("add-system").addEventListener("click", () => {
```

Comment:
This is a pretty big function, might want to split it up and make sure each function does one thing.